### ISSUE TITLE
WCF streaming: the response stream loses the last 1 byte whenever the returned stream length is a multiple of 384

### ISSUE THREAD

**Issue by KKhurin** [dotnet/wcf#267]
This is a functional bug (and it was discovered while writing stress tests). We simply iterate through all values of data in:
~~~
        [OperationContract]
        Stream GetStreamFromInt(int data);
~~~

Whenever we request a stream of size multiple of 384 we will miss the last byte of the response stream (e.g. we'll get end of stream 1 byte earler - all other bytes are correct no matter what the size of the stream is).

**Comment by roncain** [comment]
@KKhurin -- nice find.  It may be worthwhile making a normal OuterLoop scenario test that replicates this so that it is easier to debug.  It can be converted to a [Theory] based test to try a variety of sizes, both in the request and in the response.

**Comment by zhenlan** [comment]
@shmao Please take a look of this issue with high priority to see if this is an issue fixable in our code or it is an issue with HttpClient.

**Comment by shmao** [comment]
This is caused by a bug in XmlBaseReader. I've sent out a [PR](https://github.com/dotnet/corefx/pull/2909) for the fix.

**Comment by shmao** [comment]
Closing the issue as it had been fixed by dotnet/corefx#2909 .
